test_journal.py

```diff
@@ -233,18 +211,79 @@ def test_logout(app):
     redirect = app.get('/logout', status="3*")
     response = redirect.follow()
     assert response.status_code == 200
-    actual = response.body
 
 
 def test_home(app):
     test_login_success(app)
     redirect = app.get('/', status=200)
-    response = redirect.click(linkid = "home")
+    response = redirect.click(linkid="home")
     assert response.status_code == 200
 
-def test_newpost(app):
-    redirect = app.get('/newpost', status=200)
+
+def test_post_logout(app):
+    entry_data = {
+        'title': 'Hello there',
+        'text': 'This is a post',
+    }
+    response = app.post('/add', params=entry_data, status='3*')
+    redirected = response.follow()
+    actual = redirected.body
+    for expected in entry_data.values():
+        assert 'No entries here so far' in actual
+
+
+def test_post_login(app, auth_req):
+    redirect = app.get('/', status=200)
+    response = redirect.click(linkid="login")
+    username, password = ('admin', 'secret')
+    redirect = login_helper(username, password, app)
+    assert redirect.status_code == 302
+    response = redirect.follow()
+    assert response.status_code == 200
+    entry_data = {
+        'title': 'Hello there',
+        'text': 'This is a post',
+    }
+    response = app.post('/add', params=entry_data, status='3*')
+    redirected = response.follow()
+    actual = redirected.body
+    for expected in entry_data.values():
+        assert expected in actual
 
 
+def test_post_markdown_nocode(app, auth_req):
+    redirect = app.get('/', status=200)
+    response = redirect.click(linkid="login")
+    username, password = ('admin', 'secret')
+    redirect = login_helper(username, password, app)
+    assert redirect.status_code == 302
+    response = redirect.follow()
+    assert response.status_code == 200
+    entry_data = {
+        'title': 'Hello there',
+        'text': 'This is a post'
+    }
+    response = app.post('/add', params=entry_data, status='3*')
+    redirected = response.follow()
+    actual = redirected.body
+    for expected in entry_data.values():
+        assert 'codehilite' not in actual
 
 
+def test_post_markdown_code(app, auth_req):
+    redirect = app.get('/', status=200)
+    response = redirect.click(linkid="login")
+    username, password = ('admin', 'secret')
+    redirect = login_helper(username, password, app)
+    assert redirect.status_code == 302
+    response = redirect.follow()
+    assert response.status_code == 200
+    entry_data = {
+        'title': 'Hello there',
+        'text': '    import OS'
+    }
+    response = app.post('/add', params=entry_data, status='3*')
+    redirected = response.follow()
+    actual = redirected.body
+    for expected in entry_data.values():
+        assert 'codehilite' in actual
```

Comment:
You could use the `redirected.html` attribute to get the beautiful soup object for this page, and then use that to get to the DOM structure that contains your rendered code.  That'd be a bit smoother to read, and less dependent on implementation details.